ui/test/nuxeo-aggregation-widget.test.js

```diff
@@ -55,42 +131,211 @@ const data = {
+  expect(elements.length).to.equal(lengthExpected);
+  elements.forEach((element) => {
+    expect(isElementVisible(element)).to.equal(value);
+  });
+};
+
 suite('nuxeo-checkbox-aggregation', () => {
-  setup(async () => {
-    await login();
+  suite('Empty', () => {
+    test('Should display no results message when buckets are empty', async () => {
+      const nuxeoCheckboxAggregation = await fixture(
+        html`
+          <nuxeo-checkbox-aggregation></nuxeo-checkbox-aggregation>
+        `,
+      );
+
+      const message = nuxeoCheckboxAggregation.shadowRoot.querySelector('span');
+      expect(isElementVisible(message)).to.be.true;
+      expect(message.innerHTML).to.equal('checkboxAggregation.noResults');
+    });
   });
 
-  suite('Given a nuxeo-checkbox-aggregation widget on document aggregations', () => {
-    let documentAggs;
+  suite('Non-collapsible', () => {
+    test('Should display a label when "label" is set', async () => {
+      const nuxeoCheckboxAggregation = await fixture(
+        html`
+          <nuxeo-checkbox-aggregation .data=${data} label="Some label"> </nuxeo-checkbox-aggregation>
+        `,
+      );
 
-    setup(async () => {
-      documentAggs = await fixture(html`
-        <nuxeo-checkbox-aggregation .data=${data}></nuxeo-checkbox-aggregation>
-      `);
+      const label = nuxeoCheckboxAggregation.shadowRoot.querySelector('label');
+      expect(label.innerHTML).to.equal('Some label');
     });
 
-    test('It displays the title of the document', () => {
-      const choices = dom(documentAggs.root).querySelectorAll('paper-checkbox');
-      expect(choices.length).to.be.equal(2);
-      expect(choices[0].textContent.trim()).to.be.equal('Tolkien (2)');
-      expect(choices[1].textContent.trim()).to.be.equal('Asimov (1)');
+    test('Should display the title of the document when "data" is set', async () => {
+      const nuxeoCheckboxAggregation = await fixture(
+        html`
+          <nuxeo-checkbox-aggregation .data=${data}></nuxeo-checkbox-aggregation>
+        `,
+      );
+
+      const checkboxes = nuxeoCheckboxAggregation.shadowRoot.querySelectorAll('paper-checkbox');
+      expect(checkboxes.length).to.equal(9);
+      checkboxes.forEach((element) => {
+        expect(isElementVisible(element)).to.be.true;
+      });
+      expect(checkboxes[0].innerHTML.trim()).to.equal('Tolkien (2)');
+      expect(checkboxes[1].innerHTML.trim()).to.equal('Asimov (1)');
+      expect(checkboxes[2].innerHTML.trim()).to.equal('Hemingway (3)');
+      expect(checkboxes[3].innerHTML.trim()).to.equal('Dostoevsky (4)');
+      expect(checkboxes[4].innerHTML.trim()).to.equal('Tolstoy (5)');
+      expect(checkboxes[5].innerHTML.trim()).to.equal('Pessoa (6)');
+      expect(checkboxes[6].innerHTML.trim()).to.equal('Balzac (7)');
+      expect(checkboxes[7].innerHTML.trim()).to.equal('Cervantes (8)');
+      expect(checkboxes[8].innerHTML.trim()).to.equal('Shakespeare (9)');
+    });
```

Comment:
We could make this shorter, using deep equal:
```suggestion
      expect(Array.from(checkboxes).map((c) => c.innerHTML.trim())).to.deep.equal([
        'Tolkien (2)', 'Asimov (1)', 'Hemingway (3)', 'Dostoevsky (4)', 'Tolstoy (5)', 'Pessoa (6)', 'Balzac (7)',
        'Cervantes (8)', 'Shakespeare (9)',
      ]);
```